scheduler_perf testing fail for Recorder is nil

<!-- Please only use this template for submitting reports about failing tests in Kubernetes CI jobs -->

**Which jobs are failing**:

**Which test(s) are failing**:

**Since when has it been failing**:

**Testgrid link**:

**Reason for failure**:
I0116 21:50:32.171316    3959 wrap.go:47] GET /healthz: (1.0472ms) 500
goroutine 2352 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).recordStatus(0xc001fa7110, 0x1f4)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:204 +0xd2
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog.(*respLogger).WriteHeader(0xc001fa7110, 0x1f4)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/httplog/httplog.go:183 +0x35
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*baseTimeoutWriter).WriteHeader(0xc00382ad80, 0x1f4)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:205 +0xaf
net/http.Error(0xdc0d5b8, 0xc0036a21c0, 0xc0003b5a40, 0x136, 0x1f4)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1976 +0xda
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz.handleRootHealthz.func1(0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/healthz/healthz.go:184 +0x3aa
net/http.HandlerFunc.ServeHTTP(0xc0027a40e0, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*pathHandler).ServeHTTP(0xc00278a900, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:241 +0x510
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux.(*PathRecorderMux).ServeHTTP(0xc00014f340, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/mux/pathrecorder.go:234 +0x8a
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server.director.ServeHTTP(0x4c243b1, 0xe, 0xc0008aa360, 0xc00014f340, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/handler.go:154 +0x661
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthorization.func1(0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authorization.go:64 +0x4c3
net/http.HandlerFunc.ServeHTTP(0xc0000da080, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.WithMaxInFlightLimit.func1(0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/maxinflight.go:160 +0x459
net/http.HandlerFunc.ServeHTTP(0xc00071c9c0, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithImpersonation.func1(0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go:50 +0x1eeb
net/http.HandlerFunc.ServeHTTP(0xc0000da700, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ef00)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters.WithAuthentication.func1(0xdc0d5b8, 0xc0036a21c0, 0xc00380ee00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/endpoints/filters/authentication.go:81 +0x476
net/http.HandlerFunc.ServeHTTP(0xc0008a9400, 0xdc0d5b8, 0xc0036a21c0, 0xc00380ee00)
	/usr/local/Cellar/go/1.11.1/libexec/src/net/http/server.go:1964 +0x44
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1(0xc003637560, 0xc00045dd00, 0x6b6d240, 0xc0036a21c0, 0xc00380ee00)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0xb3
created by k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:97 +0x1b0

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x25f7a93]

goroutine 3306 [running]:
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).bind(0xc00002db60, 0xc004ae3c00, 0xc0049d0a80, 0x0, 0x1c4efe1)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:428 +0x643
k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne.func1(0xc001ffe701, 0xc00002db60, 0xc004ae3c00, 0x6b51d40, 0xc003a1c1c0, 0xc001ffe700, 0x11, 0x64, 0x64, 0xbf07eb8a4781e3e8, ...)
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:556 +0x21f
created by k8s.io/kubernetes/pkg/scheduler.(*Scheduler).scheduleOne
	/Users/wangguoliang/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/scheduler/scheduler.go:521 +0xb6f
exit status 2
FAIL	k8s.io/kubernetes/test/integration/scheduler_perf	4.409s
!!! Error in ./test-performance.sh:51
  Error in ./test-performance.sh:51. 'go test -test.run=. -test.timeout=60m -test.short=false' exited with status 1
Call stack:
  1: ./test-performance.sh:51 main(...)
Exiting with status 1
~/Documents/mos/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_perf

**Anything else we need to know**:
run shell
`kubernetes/test/integration/scheduler_perf/test-performance.sh`

--- a/test/integration/util/BUILD
+++ b/test/integration/util/BUILD
@@ -13,6 +13,7 @@ go_library(
     ],
     importpath = "k8s.io/kubernetes/test/integration/util",
     deps = [
+        "//pkg/api/legacyscheme:go_default_library",
         "//pkg/cloudprovider/providers/gce:go_default_library",
         "//pkg/scheduler:go_default_library",
         "//pkg/scheduler/api:go_default_library",

--- a/test/integration/util/util.go
+++ b/test/integration/util/util.go
@@ -26,6 +26,7 @@ import (
 	clientv1core "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/scheduler"
 	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
 	"k8s.io/kubernetes/pkg/scheduler/factory"
@@ -70,6 +71,7 @@ func StartScheduler(clientSet clientset.Interface) (factory.Configurator, Shutdo
 	if err != nil {
 		klog.Fatalf("Error creating scheduler: %v", err)
 	}
+	config.Recorder = evtBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"})
 
 	sched := scheduler.NewFromConfig(config)
 	informerFactory.Start(stopCh)

